node/src/identity/hsmsigners.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2021 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as pkcs11js from 'pkcs11js';
+import * as elliptic from 'elliptic';
+import BN = require("bn.js");
+const ecSignature = require('elliptic/lib/elliptic/ec/signature.js');  // eslint-disable-line
+import { Signer } from './signer';
+
+type DER = string | number[];
+
+export interface HSMSignerOptions {
+    /**
+     * the path to the PKCS11 library
+     */
+    library: string;
+
+    /**
+     * The label associated with the token for the slot
+     */
+    label: string;
+
+    /**
+     * The pin for the slot identified by the label
+     */
+    pin: string;
+
+    /**
+     * Identifier. The CKA_ID assigned to the HSM object
+     */
+    identifier: string | Buffer;
+
+    /**
+     * Optional user type for the HSM. If not specified it defaults to CKU_USER
+     */
+    userType?: number;
+}
+
+export type HSMClose = () => void;
+export type HSMSigner = {signer: Signer, close: HSMClose};
+
+let hsmSignerFactory: HSMSignerFactory | undefined;
+
+/**
+ * Create a new HSM signing implementation based on provided HSM options.
+ *
+ * This returns an object with 2 properties
+ * - signer which is the signer function
+ * - close which is a close function to close the signer when it's not required anymore
+ *
+ * @param hsmSignerOptions - The HSM signer options
+ * @returns an HSM Signer implementation
+ */
+export const newHSMSigner = (hsmSignerOptions: HSMSignerOptions): HSMSigner => {
+    if (!hsmSignerFactory) {
```

Comment:
This looks really dangerous. So the following code would end up with both signers unexpectedly using LIB_1?
```
const signer1 = newHSMSigner({ lib: 'LIB_1', ... });
const signer2 = newHSMSigner({ lib: LIB_2', ... });
```

It seems like we should either:
1. Create a new HSMSignerFactory each time; or
2. If creating multiple HSMSigner instances from the same factory is important (i.e. the PKCS11 load/initialize must only happen once), expose the HSMSignerFactory for the client to create and use to create HSM signers as required.